### ISSUE TITLE
Bug: no-restricted-imports doesn't allow for lookahead/lookbehind and their negative versions in the importNamePattern

### ISSUE THREAD

**Issue by julianobrasil** [eslint/eslint#19132]
### Environment

Node version: v20.10.0
npm version: v10.4.0
Local ESLint version: v9.12.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.1.0

### What parser are you using?

Other

### What did you do?

<details>
<summary>Configuration</summary>

```
'no-restricted-imports': [
  'error',
  {
    patterns: [
      {
         group: ['@app/utils'],
         importNamePattern: '?!Abcd$',
         message: 'You can only import names that end with Abcd from @app/utils', 
      },
    ],
  },
],
```
</details>

```js
import {
  z, // this one should cause a eslint error
  sumAbcd,
  fnAbcd
} from '@app/utils'
```

[Reproduction link](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcmVzdHJpY3RlZC1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IHBhdHRlcm5zOiBbe1xuICAgIFwiaW1wb3J0TmFtZVBhdHRlcm5cIjogXCI/IShBYmNkKVwiLFxuICAgIFwiZ3JvdXBcIjogW1wiQGFwcC91dGlsc1wiIF1cbn1dfV0qL1xuXG4vLyBlcnJvclxuaW1wb3J0IHsgeiB9IGZyb20gJ0BhcHAvdXRpbHMnO1xuXG4vLyBubyBlcnJvclxuaW1wb3J0IHsgdmFyQWJjZCB9IGZyb20gJ0BhcHAvdXRpbHMnO1xuIiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJzb3VyY2VUeXBlIjoibW9kdWxlIiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)

### What did you expect to happen?

I expected to be able to use any regex allowed by JS syntax

### What actually happened?

The lookahead and lookbehind (and their negative versions) are not recognized as valid regexes

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tcmVzdHJpY3RlZC1pbXBvcnRzOiBbXCJlcnJvclwiLCB7IHBhdHRlcm5zOiBbe1xuICAgIFwiaW1wb3J0TmFtZVBhdHRlcm5cIjogXCI/IShBYmNkKVwiLFxuICAgIFwiZ3JvdXBcIjogW1wiQGFwcC91dGlsc1wiIF1cbn1dfV0qL1xuXG4vLyBlcnJvclxuaW1wb3J0IHsgeiB9IGZyb20gJ0BhcHAvdXRpbHMnO1xuXG4vLyBubyBlcnJvclxuaW1wb3J0IHsgdmFyQWJjZCB9IGZyb20gJ0BhcHAvdXRpbHMnO1xuIiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJzb3VyY2VUeXBlIjoibW9kdWxlIiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by julianobrasil** [comment]
It seems we already have a way out for it with the `allowImportNamePattern`. I'll reopen this one if needed

**Comment by mdjermanovic** [comment]
`?!(Abcd)` is an invalid pattern. Valid syntax for negative lookahead is `(?!Abcd)`.
